go-controller/pkg/ovn/policy.go

```diff
@@ -67,17 +70,47 @@ type networkPolicy struct {
 	svcHandlerList  []*factory.Handler
 	nsHandlerList   []*factory.Handler
 
-	// localPods is a list of pods affected by this policy
-	// this is a sync map so we can handle multiple pods at once
-	// map of string -> *lpInfo
-	localPods sync.Map
-
+	// portGroupName is the name of the shared port group this network policy belongs to
 	portGroupName string
 	deleted       bool //deleted policy
 	created       bool
+
+	// indicate if the policy needs ingress/egress default deny rule
+	ingressDefDeny bool
+	egressDefDeny  bool
+
+	// sharePortGroupName is in the form of <policy_namespace>_shared_port_group_<localPodSelector.String>
+	// it is used as the key in each per-namespace spgInfoMap (oc.spgInfoMap[namespace]), and it is also used as
+	// the external-ids:name field of the associated shared port group
+	sharePortGroupName string
+}
+
+// now we only support portGroup sharing if the policy's local pod selector is empty. This can be changed later.
+func getSharedPortGroupName(policy *knet.NetworkPolicy) string {
+	sel, _ := metav1.LabelSelectorAsSelector(&policy.Spec.PodSelector)
+	return policy.Namespace + "_" + "shared_port_group" + "_" + sel.String()
```

Comment:
I guess you can just use `FormatLabelSelector` that will return string for given selector, but I am concerned that `sel.String()` may return different strings for equal selectors (LabelSelector.MatchLabels is a map, and I don't see any sorting in String() method)
And then this string can be very long, I don't know if we can safely put it to ExternalIDs of port group, may need some db tests to check